rtv/oauth.py

```diff
@@ -99,15 +110,25 @@ def authorize(self):
         self.open_config(update=True)
         # If no previous OAuth data found, starting from scratch
         if not self.config.has_section('oauth') or not self.config.has_option('oauth', 'refresh_token'):
+            # Start HTTP server and listen on port 65000
+            self.callback_app.listen(65000)
+            
             # Generate a random UUID
             hex_uuid = uuid.uuid4().hex
 
             permission_ask_page_link = self.reddit.get_authorize_url(str(hex_uuid),
                 scope=self.scope, refreshable=True)
 
-            with self.loader(message='Waiting for authorization'):
-                webbrowser.open(permission_ask_page_link)
+            if self.compact:
+                show_notification(self.stdscr, ['Opening ' + os.environ.get('BROWSER')])
+                curses.endwin()
+                webbrowser.open_new_tab(permission_ask_page_link)
                 ioloop.IOLoop.current().start()
```

Comment:
The redirection is never done with terminal web browser, but if you close the web browser after clicking Authorize, the process goes on normally. Now it may be possible to make the redirection work, but the user will still have to close the web browser after landing on the page, since the browser runs in foreground. I'll see what I can do.